modules/control/common/mrac_controller.h

```diff
@@ -0,0 +1,154 @@
+   * gain
+   * @param mrac_conf configuration for adaption model
+   */
+  void SetAdaptionModel(const MracConf &mrac_conf);
+
+  /**
+   * @brief transfer mrac (1st or 2nd) order reference model coefficients to the
+   discrete-time form, with the bilinear transform (trapezoidal integration)
+   method
+   * @param dt sampling time interval
+   */
+  void TransformReferenceModel(const double dt);
+
+  /**
+   * @brief exexute the adaption interation with respect to the designed law in
+   discrete-time form, with the bilinear transform (trapezoidal integration)
+   method
+   * @param law_adp adaptive law at k and k-1 steps
+   * @param state_adp state used in the adaptive law at k and k-1 steps
+   * @param gain_adp adaptive gain for the given adaptive law
+   */
+  void RunAdaptionModel(std::vector<double> *law_adp,
+                        const std::vector<double> state_adp,
+                        const double gain_adp);
+
+  /**
+   * @brief reset variables for mrac controller
+   */
+  void Reset();
+
+  /**
+   * @brief compute control value based on the error
+   * @param error error value, the difference between
+   * a desired value and a measured value
+   * @param dt sampling time interval
+   * @return control value based on mrac terms
+   */
+  virtual double Control(const double error, const double state,
+                         const double dt);
+
+  /**
+   * @brief get saturation status for reference system
+   * @return saturation status
+   */
+  int ReferenceSaturationStatus() const;
+
+  /**
+   * @brief get saturation status for control system
+   * @return saturation status
+   */
+  int ControlSaturationStatus() const;
+
+ protected:
+  // reference model as a digital filter
+  common::DigitalFilter reference_model_;
+  // flag to indicate if the reference model is valid
+  bool ref_model_enabled_ = false;
+  // The order of the reference model
+  int reference_model_order_ = 1;
+  // 1st-order Reference system coefficients in contiouous-time domain
+  double tau_ref_ = 0.0;
+  // 2nd-order Reference system coefficients in contiouous-time domain
+  double wn_ref_ = 0.0;
+  double zeta_ref_ = 0.0;
+  double Ts_ = 0.01;  // By default, control sampling time is 0.01 sec
+
+  // Adaption system coefficients
+  double gamma_x_adp_;      // State adaption gain
+  double gamma_r_adp_;      // Desired command adaption gain
+  double gamma_theta_adp_;  // Nonlinear dynamics adaption gain
+
+  // Reference system coefficients in discrete-time domain
+  std::vector<double> kd_ref_{0.0, 0.0, 0.0};  // Denominator coefficients
+  std::vector<double> kn_ref_{0.0, 0.0, 0.0};  // Numerator coefficients
+
+  // Adaption system input variables in discrete-time domain
+  std::vector<double> r_des_{0.0, 0.0};       // Updated desired command vector
+  std::vector<double> x_act_{0.0, 0.0};       // Updated actuation states vector
+  std::vector<double> x_ref_{0.0, 0.0};       // Reference states vector
```

Comment:
Try to use more meaningful names instead of abbreviations. Many of them are not readable to me.

Reply:
Fixed per comment. Thanks.

Reply:
fixed per comment. Thanks